packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -959,6 +964,24 @@ export const RemoteServerFactory = {
         const server = awsServerlessExpress.createServer(app, null, binaryMimeTypes)
 
         const handler = (event, context, callback) => {
+            // encode non ASCII request headers
+            if (options?.encodeNonAsciiHttpHeaders) {
```

Comment:
So I know I was the one that originally suggested this feature flag, but I think in the latest discussion I thought we were talked about it not making sense. 

The current behaviour is that the server will crash if non-ascii characters are present. I don't think that is the desired behaviour for anyone. We had 2 options, 1 delete the headers, 2 encode them again. 

I personally like the idea of deleting them as it would motivate people to fix the source of the problem, but it looks like we are going with the later option. Do we need to default for this flag to be false?

Reply:
@bfeister want your thoughts on this comment too.

Reply:
The reason for the feature flag (and you might not have been in this discussion, I also don't think I was in the discussion you allude to in your comment) is that a customer might very specifically want to include these values and not strip them out. Imagine a grocery store like ASDA that wants to use cloudflare geolocation, they can do some workaround at the edge if we feature flag this. If we auto-strip all of them as a default, there's no way for them to use geolocation for city names with non-ASCII chars

Reply:
Also from a high-level philosophical standpoint, I just think this is categorically the kind of thing that is very likely to want to be one way or the other and outside influence can change that. So I'd prefer customers being able to toggle it at will to get themselves out of a bind

Reply:
If you're good with defaulting to "ON" and keeping the feature flag, then that works for me 👍

Reply:
Can you explain it again, maybe in simpler terms, why we need the flag? I'm just not grasping it.

Reply:
Some customers may want to convert this in some other layer in the stack. And some customers will prefer to "protect" their site from these passing through. Others will prefer to see non-english characters in a non-encoded format. Also, if the layer where the crash happens is in our control, we fix it, and then customers can't "eject" to simply have the headers unmodified for their Japanese language site, we make it hard for them to get to the state they would like